### PR TITLE
Feature6 instructor create course

### PR DESCRIPTION
**Pull request by FXXFERMI**
Finished the use case, Interfaces, interface adapters, framework,and screen.

### CODE REVIEW

File: src/main/java/Main.java
Comment (hassa963) [inline src/main/java/Main.java:35]: I think this needs to be refactored so that it's instead set up in the screen

File: src/main/java/userinterface/InstuctorCreateCourseUserInterface/CreateCourseScreen.java
Comment (hassa963) [inline src/main/java/userinterface/InstuctorCreateCourseUserInterface/CreateCourseScreen.java:46]: instead of passing in the presenter to the screen in, initialize it before you retrieve the controller 
something like this CreateCoursePresenter presenter = new CreateCoursePresenter( new DatabaseDriver());
this.controller = presenter.getController(); not a huge change but it makes our main cleaner which follows the feedback we got from pritish

**Review by hassa963 — COMMENTED** [review]
I would make that change I mentioned about the screen as well as add your documentation  and tests before merging but looks good I would also wait for everyone else's feedback.

File: src/main/java/frameworksdrivers/Gateways/CourseDbGateway.java
Comment (hassa963) [inline src/main/java/frameworksdrivers/Gateways/CourseDbGateway.java:138]: This method doesn't need a change as far as I know, however as we haven't changed how we changed  what we save to sessionstudent table there will be a bug slightly in course enrollment but we can fix this for our final implementation.

File: src/main/java/Main.java
Comment (FXXFERMI) [reply src/main/java/Main.java:39]: You mean line 32,33,34and 35 or just line35?

File: src/main/java/interfaceadapters/CreateCourseInterfaceAdapters/SuccessCreate.java
Comment (jpmedinagl) [inline src/main/java/interfaceadapters/CreateCourseInterfaceAdapters/SuccessCreate.java:9]: I think this class is not required. Because exceptions are only ever attributed to "exceptional failures" that the code is hoping to catch. A success creation is what we expect, so we shouldn't throw an error. This class isn't used anywhere in the code so removing it wouldn't cause any issues.

File: src/main/java/usecases/InstructorCreateCourseUseCase/CreateCourseInputBoundary.java
Comment (jpmedinagl) [inline src/main/java/usecases/InstructorCreateCourseUseCase/CreateCourseInputBoundary.java:1]: I see that the documentation is very well done across all the files. Good job !

File: src/main/java/usecases/InstructorCreateCourseUseCase/CreateCourseInteractor.java
Comment (jpmedinagl) [inline src/main/java/usecases/InstructorCreateCourseUseCase/CreateCourseInteractor.java:45]: I really like the way that these two methods are abstracted and are written very well.

**Review by jpmedinagl — APPROVED** [review]
Well done. I see that you've done a throughout job doing tests and javadocs. Everything looks good on my end.

I've left some additional comments that you might reconsider refactoring at a later stage. But this working version is well done !

**Review by jpmedinagl — APPROVED** [review]
I just pulled from main in this branch so that it would be up to date since it showed merge conflicts in both Main and CourseDbGateway.

**Pull request merged**
